starknet_py/tests/e2e/client/gateway_test.py

```diff
@@ -47,3 +50,53 @@ async def test_get_transaction_status(invoke_transaction_hash, gateway_client):
     assert isinstance(tx_status_resp, TransactionStatusResponse)
     assert tx_status_resp.transaction_status == TransactionStatus.ACCEPTED_ON_L2
     assert isinstance(tx_status_resp.block_hash, int)
+
+
+# pylint: disable=protected-access
+@pytest.mark.parametrize(
+    "net, chain",
+    ((TESTNET, StarknetChainId.TESTNET), (MAINNET, StarknetChainId.MAINNET)),
+)
+def test_creating_client_from_predefined_network(net, chain):
+    gateway_client = GatewayClient(net=net)
+    net_address = net_address_from_net(net)
+
+    assert gateway_client.net == net
+    assert gateway_client._feeder_gateway_client.url == f"{net_address}/feeder_gateway"
+    assert gateway_client._gateway_client.url == f"{net_address}/gateway"
+    assert gateway_client.chain == chain
```

Comment:
I think in this case it'd be better to use values for urls directly rather than using `net_address_from_net`. If there was an error in the `net_address_from_net`, we wouldn't notice.